clang/lib/Sema/BoundsAnalysis.cpp

```diff
@@ -147,34 +150,47 @@ void BoundsAnalysis::ComputeGenSets() {
       // default case for a switch) is always the last edge in the list of
       // edges. So we check that EB is not on the last edge for pred.
 
-      // TODO: Allow bounds widening for the default case of a switch-case.
-      // If we establish that another label in a switch statement tests for 0,
-      // then the default case will handle non-zero case, and the bounds can be
-      // widened there. The following github issue tracks this:
-      // https://github.com/microsoft/checkedc-clang/issues/818.
-
-      if (pred->succ_size() && EB->Block != *(pred->succs().end() - 1)) {
-	// Get the edge condition and fill the Gen set.
-	if (Expr *E = GetTerminatorCondition(pred)) {
-
-          // Check if the pred ends in a switch statement.
-          if (isa<SwitchStmt>(pred->getTerminatorStmt())) {
-            // We can widen the bounds only if the current block has a non-null
-            // case statement.
-            if (IsSwitchCaseNull)
-              continue;
-
-	    // If the switch expression is integral, strip off the
-	    // IntegralCast.
-	    if (auto *CE = dyn_cast<CastExpr>(E)) {
-              if (CE->getCastKind() == CastKind::CK_IntegralCast)
-                E = CE->getSubExpr();
-            }
-          }
+      if (pred->succ_size() == 0)
+        continue;
+
+      const CFGBlock *FalseOrDefaultBlock = *(pred->succs().end() - 1);
+      if (EB->Block == FalseOrDefaultBlock)
+        continue;
+
+      // Get the edge condition.
+      Expr *E = GetTerminatorCondition(pred);
+      if (!E)
+        continue;
 
-          FillGenSet(E, BlockMap[pred], EB);
+      // Check if the pred ends in a switch statement.
+      const Stmt *TerminatorStmt = pred->getTerminatorStmt();
+      if (TerminatorStmt && isa<SwitchStmt>(TerminatorStmt)) {
+
+        // According to C11 standard section 6.8.4.2, the controlling
+        // expression of a switch shall have integer type.
+        // If we have switch(*p) where p is _Nt_array_ptr<char> then it is
+        // casted to integer type and an IntegralCast is generated. Here we
+        // strip off the IntegralCast.
+        if (auto *CE = dyn_cast<CastExpr>(E)) {
+          if (CE->getCastKind() == CastKind::CK_IntegralCast)
+            E = CE->getSubExpr();
+        }
+
+        // If the current block has a non-null case label, we cannot widen the
```

Comment:
The comment says "If the current block has a non-null case label", but the statement checks whether the case is null. Should the comment say "If the current block has a null case label"?